cluster-autoscaler/cloudprovider/juju/juju_cloud_provider.go

```diff
@@ -138,22 +153,48 @@ func BuildJuju(
 		defer configFile.Close()
 	}
 
-	man := &Manager{
-		units:    make(map[string]*Unit),
-	}
-	man.init()
-
-	ng := &NodeGroup{
-		id:         "juju",
-		minSize:    3,
-		maxSize:    10,
-		target:     len(man.units),
-		manager:    man,
+	ngs := []cloudprovider.NodeGroup{}
+	for _, nodeGroupSpecString := range do.NodeGroupSpecs {
+		nodeGroupSpec, err := dynamic.SpecFromString(nodeGroupSpecString, scaleToZeroSupported)
```

Comment:
Ahh, i see [this](https://github.com/charmed-kubernetes/autoscaler/blob/master/cluster-autoscaler/config/dynamic/node_group_spec.go#L38) splits a string 3 times with ':' probably leading to why you chose `min:max:node-group`  where `node-group` is split again with a `,`

Reply:
However `strings.SplitN(nodeGroupSpecString, 3)` will only split three times -- leaving the rest of the string unsplit.  
if you'd be willing to entertain that 3rd string token here can be FULL of colons (`:`) then i'd propose we continue using it as a delimiter for `<model>:<app>`   i don't think `:` can be used in the name of a juju application